### PR TITLE
Autocomplete: update to @textcomplete package and close on blur

### PR DESCRIPTION
**Pull request by MaxLeiter**
The library we use, https://github.com/yuku/old-textcomplete, was archived in favor of this new (almost) drag-and-drop replacement: https://github.com/yuku/textcomplete. I also addressed #1085 as it was a simple fix.

### CODE REVIEW

File: client/js/autocompletion.js [hunk 1]
```diff
@@ -214,29 +216,21 @@ function enableAutocomplete(input) {
...
 		emojiStrategy,
 		nicksStrategy,
 		chanStrategy,
 		commandStrategy,
 		foregroundColorStrategy,
 		backgroundColorStrategy,
-	]);
+	];
 
-	// Activate the first item by default
```
Comment (MaxLeiter) [inline client/js/autocompletion.js:234]: Having tested on 4.3.0, I don't think this does anything anymore, and the issue is hidden / not archived on Wayback Machine, so I removed this code. The Textcomplete types don't point to `dropdown` valid anymore.

File: client/js/autocompletion.js [hunk 2]
```diff
@@ -3,7 +3,9 @@
 const constants = require("./constants");
 
 import Mousetrap from "mousetrap";
-import {Textcomplete, Textarea} from "textcomplete";
+const {Textcomplete} = require("@textcomplete/core");
+const {TextareaEditor} = require("@textcomplete/textarea");
```
Comment (itsjohncs) [inline client/js/autocompletion.js:7]: Feels odd to mix require and import here. Does this pattern work instead?:

```js
import textcomplete from "@textcomplete/core";
const {Textcomplete} = textcomplete;
```

**Review by itsjohncs — APPROVED** [review]
LGTM. Tested locally (briefly) on Safari on Mac OS and the emoji picker works as before afaict.

File: client/js/autocompletion.js (hunk 2 above)
Comment (MaxLeiter) [reply client/js/autocompletion.js:7]: yeah this will work, I just used vscodes auto import tool and didn't bother checking.

**Pull request merged**